File: Hayaa.BaseModel/Hayaa.BaseModel/Model/GridPager/GridPager.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Hayaa.BaseModel
{
    /// <summary>
    /// 分页数据容器类,2147483647数据量内使用此类
    /// </summary>
    [Serializable]
    public class GridPager<T>: FunctionListResult<T> where T:BaseData
    {
        /// <summary>
        /// 当前指向数据页
        /// </summary>
        public int CurrentIndex { set; get; }
        /// <summary>
        /// 数据页尺寸
        /// </summary>
        public int PageSize { set; get; }
        /// <summary>
        ///数据总量
        /// </summary>
        public int Total { set; get; }
       
    }
}

File: Java/hayaa.basemodel/src/main/java/hayaa/basemodel/model/GridPager/GridPager.java
package hayaa.basemodel.model.GridPager;

import hayaa.basemodel.model.BaseData;
import hayaa.basemodel.model.FunctionListResult;

public class GridPager<T extends BaseData> extends FunctionListResult<T> {
    private int CurrentIndex;
    private int PageSize;
    private int Total;
    public GridPager(int current,int size){
        this.CurrentIndex=current;
        this.PageSize=size;
    }
    public int getCurrentIndex() {
        return CurrentIndex;
    }

    public void setCurrentIndex(int currentIndex) {
        CurrentIndex = currentIndex;
    }

    public int getPageSize() {
        return PageSize;
    }

    public void setPageSize(int pageSize) {
        PageSize = pageSize;
    }

    public int getTotal() {
        return Total;
    }

    public void setTotal(int total) {
        Total = total;
    }


}
